clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,506 @@
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+
+	// Create clusterRoleBinding
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating clusterRoleBinding: %v", err)
+	}
+
+	// Create CRD
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	// Delete CRD
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	// Delete clusterRoleBinding
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	npm.templateMapping = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
+	if err := npm.framework.ApplyTemplatedManifests(workerPodDeploymentManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating worker pods :%v ", err)
+	}
+
+	// Wait for all worker pods to be ready
+	weightedPodReadyTimeout := npm.getWeightedPodReadyTimeout()
+	stopCh := make(chan struct{})
+	time.AfterFunc(weightedPodReadyTimeout, func() {
+		close(stopCh)
+	})
+	options := &measurementutil.WaitForPodOptions{
+		Selector:            &measurementutil.ObjectSelector{Namespace: netperfNamespace},
+		DesiredPodCount:     npm.numberOfClients + npm.numberOfServers,
+		CallerName:          networkPerformanceMetricsName,
+		WaitForPodsInterval: podReadyPollInterval,
+	}
+	if err := measurementutil.WaitForPods(npm.k8sClient, stopCh, options); err != nil {
+		return err
+	}
+	return nil
+}
+
+// getWeightedTimerValuesForPoll returns poll timeout value.
+// Pod numbers are taken into consideration to decide the timer values.
+func (npm *networkPerformanceMeasurement) getWeightedPodReadyTimeout() time.Duration {
+	weightedPodReadyTimeout := float32(npm.numberOfServers+npm.numberOfClients) * podReadyTimeOutFactor
```

Comment:
I'm not a fan of those magic constants. The fact that we need to download images makes it quite unpreditible.

So instead of that, I would much rather prefer setting it e.g. to constant 5m with a TODO to figure out a potentially better timeout when needed.

Reply:
Since we have preload image feature available , downloading image should not be a problem. Making a wait time as 5m might increase the total execution time of the testcase. I think this weighted time works well. We wanted to fine tune this further based on current logic itself.

Reply:
> Making a wait time as 5m might increase the total execution time of the testcase.

Only if something isn't starting correctly, which generally suggests some problems.
So I'm actually opposed to using a set of magic constants.